lib/rucio/core/request.py

```diff
@@ -163,6 +156,30 @@ def queue_requests(requests, session=None):
             for request in query_existing_requests:
                 existing_requests.append(request)
 
+    request_scopes_names = [(request['scope'], request['name']) for request in requests]
+    results = session.query(models.RSEFileAssociation.scope, models.RSEFileAssociation.name, models.RSEFileAssociation.rse_id, models.Distance.dest_rse_id, models.Distance.ranking)\
```

Comment:
So this query is basically to fill the `src_rse_id` since we do not have it at this point.
I am wondering if we should do this for all requests, or only the ones affected by the throttler, but it is probably good to do it just for all. I hope that the query does not slow down the workload too much, but that we have to see.
Som change requests:
- Please add chunking here (Look at other modules where this is done) The requests list can potentially be quite long, so you would be querying for thousands of replicas in one query. I wouldn't do chunks larger than 50 here.
- Right now you are selecting a random source, but we should at least make sure that a disk source is selected, if it is available (and not a tape)